frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -0,0 +1,229 @@
+package dockerfile2llb
+
+import (
+	"bytes"
+	"context"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/builder/dockerfile/instructions"
+	"github.com/docker/docker/builder/dockerfile/parser"
+	"github.com/moby/buildkit/client/llb"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	emptyImageName = "scratch"
+)
+
+type ConvertOpt struct {
+	Target       string
+	MetaResolver llb.ImageMetaResolver
+}
+
+func Dockerfile2LLB(ctx context.Context, dt []byte, opt ConvertOpt) (*llb.State, error) {
+	if len(dt) == 0 {
+		return nil, errors.Errorf("the Dockerfile cannot be empty")
+	}
+
+	dockerfile, err := parser.Parse(bytes.NewReader(dt))
+	if err != nil {
+		return nil, err
+	}
+
+	stages, metaArgs, err := instructions.Parse(dockerfile.AST)
+	if err != nil {
+		return nil, err
+	}
+
+	_ = metaArgs
```

Comment:
what is this?

Reply:
this is "args-in-from", currently unhandled like the rest of args/variable expansion.